test/indexer/test_hca_indexer.py

```diff
@@ -96,6 +97,8 @@ def test_indexing(self):
         hits = self._get_all_hits()
         self.assertElasticsearchResultsEqual(expected_hits, hits)
 
+    http.client._MAXHEADERS = 1000  # Prevents `test_deletion` to fail; https://stackoverflow.com/questions/23055378
```

Comment:
I understand the motivation (the comment makes that clear) but think about why it isn't a good idea to do this here.

Reply:
Yea my conception here was to propagate the patch at the highest context possible in this test suite. But as you hinted, only `test_deletion` requires the patch, so it is not necessary to have it at the class level 👍. Right?

Reply:
For one, this isn't a test suite. It's a test *case*. But the way this patch is done here has consequences beyond the test case. What are they? And why are they bad?

Reply:
> What are they? And why are they bad?

Since `http` is the module that defines classes that implement the client of the `HTTP` and `HTTPS` protocols it would change the default configuration.  And since this module is used by other modules/libraries this change would be propagated through the modules that use `http` and those modules would also be affected by this change. Allowing for unwanted behavior to manifest it self in other parts of the codebase unrelated to this test case.